force-app/main/default/classes/DMLOperations.cls

```diff
@@ -139,17 +143,40 @@ public with sharing class DMLOperations {
      * Question 8
      * Insert or updates opportunities from list of names.
      * Associate the Opportunities with account that has the name accountName.
-     * There should be an Opportunity for each name provided in the list.
+     * There needs to be an Opportunity for each name provided in the list.
      * HINT: The account may or may not exists in the database but is required for opportunity creation. Consider how upsert operation can help with this.
      * @param accountName The name of the Account.
      * @param oppNames The list of Opportunity names.
      */
     public static void upsertOpportunities(String accountName, List<String> oppNames) {
-        // Loop through each Opportunity name
-            // Create an Opportunity with the given name
-            // Add the Opportunity to the list
+        List <Account> Accounts = [SELECT Id FROM Account WHERE Name = :accountName]; // Query the account
+        Account varAccount = new Account(Name=accountName);
+        if(Accounts.isempty()){
+        insert varAccount;}
+        else {varAccount=[SELECT Id FROM Account WHERE Name = :accountName];}    
```

Comment:
You already queries the accounts, you don't need to query again. You can just get the first item from your list.